test/async-hooks/test-no-mix-contexts.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { AsyncContext } = require('async_hooks');
+
+const asyncContext = new AsyncContext();
+
+setTimeout(() => {
+  asyncContext.enter((store) => {
+    store.set('hello', 'world');
+    setTimeout(() => {
+      assert.strictEqual(asyncContext.getStore().get('hello'), 'world');
```

Comment:
The test could be expanded here to add a second property within this particular context... then show in the second block below that that second property does not exist... to verify that the data truly is scoped to the asyncContext.enter()

Reply:
Will do!